aiida_common_workflows/workflows/bands/quantum_espresso/generator.py

```diff
@@ -34,19 +34,20 @@ def _construct_builder(self, **kwargs) -> engine.ProcessBuilder:
         bands_kpoints = kwargs['bands_kpoints']
 
         # Find the `PwCalculation` that created the `parent_folder` and obtain the restart builder.
-        parent_calc = parent_folder.get_incoming(link_type=LinkType.CREATE).one().node
+        parent_calc = parent_folder.creator
         if parent_calc.process_type != 'aiida.calculations:quantumespresso.pw':
             raise ValueError('The `parent_folder` has not been created by a `PwCalculation`.')
-        builder_calc = parent_calc.get_builder_restart()
-
-        builder_common_bands_wc = self.process_class.get_builder()
-        builder_calc.pop('kpoints')
-        builder_common_bands_wc.pw = builder_calc
+        builder = self.process_class.get_builder()
+        
         parameters = builder_common_bands_wc.pw.parameters.get_dict()
```

Comment:
few more instances of `builder_common_bands_wc` lying around. Would be good if you could try to run it at least once in absence of unit tests.

Reply:
Huh, I think this is only in e81a8e2, in the latest commit (2e298cb) this should be fixed. It seems I was a little careless in accepting [the reorganisation of the builder code](https://github.com/aiidateam/aiida-common-workflows/pull/257#discussion_r784201566) however, I fixed this in c1d0695. Now I tested the relaxation + bands work chain for Si, and it seems to be running fine again.

Reply:
I must have seen it after that first commit. Then I haven't said anything :D